src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/SaveTempDataPropertyFilter.cs

```diff
@@ -44,6 +46,57 @@ public void OnTempDataSaving(ITempDataDictionary tempData)
             }
         }
 
+        /// <summary>
+        /// Loads and tracks any changes to the properties of the <paramref name="subject"/>.
+        /// </summary>
+        /// <param name="subject">The properties of the subject are loaded and tracked. May be a <see cref="Controller"/>.</param>
+        /// <param name="tempData">The <see cref="ITempDataDictionary"/>.</param>
+        /// <returns></returns>
+        public IDictionary<PropertyInfo, object> LoadAndTrackChanges(object subject, ITempDataDictionary tempData)
```

Comment:
This is just moving the code around without improving the design. The issue that needs to be solved isn't that we have an extra class, it's that we have two ways of doing the same thing. Let's discuss how all of this stuff is light